### PR TITLE
Add integration tests for all of the extension tomls

### PR DESCRIPTION
**Pull request by davejcameron**
### WHY are these changes introduced?

Want to make sure we have sufficient coverage over the current and new format of extension toml files so that regressions aren't introduced.

### WHAT is this pull request doing?

Testing the following
- [x] Flow trigger
- [x] Flow action
- [x] Function
- [x] UI extension
- [x] Checkout Post Purchase extension
- [x] Customer Accounts UI extension
- [x] Tax calculation extension
- [x] POS UI extension
- [x] Action extension
- [x] Web pixel
- [x] Legacy Checkout UI extension
- [x] Product subscription

### How to test your changes?

Just tests

### Post-release steps

N/A

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/app/src/cli/models/app/loader.test.ts [hunk 1]
```diff
@@ -758,6 +758,619 @@ automatically_update_urls_on_dev = true
     expect(myFunction.entrySourceFilePath).toContain(joinPath(blockPath('my-function'), 'src', 'index.js'))
   })
 
+  test('loads the app with a Flow trigger extension that has a full valid configuration', async () => {
+    // Given
+    await writeConfig(appConfiguration)
+
+    const blockConfiguration = `
+      description = "Your description"
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/app/loader.test.ts:766]: `description` would be at the `[[extensions]]` level.

File: packages/app/src/cli/models/app/loader.test.ts [hunk 2]
```diff
@@ -758,6 +758,619 @@ automatically_update_urls_on_dev = true
...
+
+      [[extensions]]
+      name = "my-flow-trigger"
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/app/loader.test.ts:769]: nit: `name` should be sentence cased (e.g. Auction bid placed)

File: packages/app/src/cli/models/app/loader.test.ts [hunk 3]
```diff
@@ -758,6 +758,619 @@ automatically_update_urls_on_dev = true
...
+      handle = "handle1"
+
+      [settings]
+
+        [[settings.fields]]
+        type = "customer_reference"
+
+        [[settings.fields]]
+        type = "single_line_text_field"
+        key = "your-field-key"
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/app/loader.test.ts:780]: nit: `key` can't have dashes, only alphanumeric chars and spaces

File: packages/app/src/cli/models/app/loader.test.ts [hunk 4]
```diff
@@ -758,6 +758,619 @@ automatically_update_urls_on_dev = true
...
+      })
+    }
+  })
+
+  test('loads the app with a Flow action extension that has a full valid configuration', async () => {
+    // Given
+    await writeConfig(appConfiguration)
+
+    const blockConfiguration = `
+      description = "Your description"
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/app/loader.test.ts:820]: Same for `description` here.

File: packages/app/src/cli/models/app/loader.test.ts [hunk 5]
```diff
@@ -758,6 +758,619 @@ automatically_update_urls_on_dev = true
...
+
+      [[extensions]]
+      name = "my-flow-action"
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/app/loader.test.ts:823]: Same here (e.g. "Place Auction bid")

**Review by MaxLeduc — COMMENTED** [review]
LGTM for the flow action and trigger. Left a couple of small comments.

File: packages/app/src/cli/models/app/loader.test.ts [hunk 6]
```diff
@@ -758,6 +758,617 @@ automatically_update_urls_on_dev = true
...
+      handle = "handle1"
+
+      [settings]
+
+        [[settings.fields]]
+        type = "customer_reference"
+
+        [[settings.fields]]
+        type = "single_line_text_field"
+        key = "your_field_key"
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/app/loader.test.ts:779]: nit: the `_` are technically invalid for the field `key` property. Only alphanumeric chars and spaces (a-Z, 0-9). Also this validation is only for triggers fields `key` property ATM which can be confusing. Actions field `key` can have non alphanumeric characters in it.

Here's the most up to date template on `main`:

```
[[extensions]]
type = "flow_trigger"
name = "{{ name }}"
handle = "{{ handle }}"
description = "Your description"

[settings]

  [[settings.fields]]
  type = "customer_reference"

  [[settings.fields]]
  type = "single_line_text_field"
  key = "Your field key"
```

Comment (jessevogt) [reply packages/app/src/cli/models/app/loader.test.ts:779]: For a metafield definition, key must match:

* regex: /\A[a-zA-Z0-9_-]+\z/ (https://github.com/Shopify/shopify/blob/main/components/content/metafields/app/models/metafield_definition.rb#L20)
* length:  min 3, max 64 (https://github.com/Shopify/shopify/blob/main/components/content/metafields/app/models/metafield_definition.rb#L45)
* must be unique under (shop_id, owner_type, owner_id, namespace)

Comment (rsperko) [reply packages/app/src/cli/models/app/loader.test.ts:779]: I believe unstructured Metafields (w/o definitions) only have the length and uniqueness constraints.

**Review by isaacroldan — APPROVED** [review]

**Pull request merged**